OgreMain/src/OgreRenderSystem.cpp

```diff
@@ -546,8 +546,8 @@ namespace Ogre {
 
             while( itor != end )
             {
-                if( bCleanManualBuffers || !(*itor)->isManual() )
-                    delete *itor;
+//                if( bCleanManualBuffers || !(*itor)->isManual() )
```

Comment:
if you revert this and apply the suggested change, we could merge this

Reply:
done. i seem to have made this more complicated than necessary though.  if it helps i can delete my branch and create a new cleaner pull request.

Reply:
no need to. This can be cleaned up on merging (squash). Even if there would be a need to re-commit, it would be cleaner to force push to this branch (overwriting), as it would link our discussion to code.